integrations/huggingface/src/index.ts

```diff
@@ -0,0 +1,115 @@
+import { interfaces } from "@botpress/sdk";
+import { HfInference } from "@huggingface/inference";
+import * as bp from ".botpress";
+import axios from "axios";
+import { generateContent } from "./actions/generateContent";
+
+export default new bp.Integration({
+  register: async ({ ctx, client, logger }) => {
+    if (!ctx.configuration.languageModels) {
+      return;
+    }
+
+    const modelIds = ctx.configuration.languageModels
+      .split(",")
+      .map((el) => el.trim());
+
+    if (modelIds.length > 10) {
+      logger.forBot().debug(`Too many models provided`);
+      return;
+    }
+    const languageModels: interfaces.llm.Model[] = [];
+    for await (const id of modelIds) {
+      try {
+        const { data }: { data: { modelId: string; pipeline_tag: string } } =
+          await axios.get(`https://huggingface.co/api/models/${id}`);
+
+        const modelId = data.modelId;
+
+        if (data.pipeline_tag !== "text-generation") {
+          logger
+            .forBot()
+            .debug(`Model ${id} is not a text generation model. Skipping...`);
+          continue;
+        }
+
+        languageModels.push({
+          id: modelId,
+          name: modelId,
+          description: modelId,
+          tags: [],
+          input: {
+            costPer1MTokens: 0,
+            maxTokens: 200000,
```

Comment:
Arbitrary number

Reply:
can we get the maxTokens it from the api result?

Reply:
Nope, we can't, we only get the input/output tokens